src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -239,10 +294,21 @@ def __str__(self):
         keyvault_dns='.vault.microsoftazure.de',
         sql_server_hostname='.database.cloudapi.de'))
 
-
 KNOWN_CLOUDS = [AZURE_PUBLIC_CLOUD, AZURE_CHINA_CLOUD, AZURE_US_GOV_CLOUD, AZURE_GERMAN_CLOUD]
 
 
+try:
+    if 'ARM_CLOUD_METADATA_URL' in os.environ:
+        arm_cloud_dict = json.loads(urlretrieve(os.getenv('ARM_CLOUD_METADATA_URL')))
+        cli_cloud_dict = convert_arm_to_cli(arm_cloud_dict)
+        if 'AzureCloud' in cli_cloud_dict:
+            cli_cloud_dict['AzureCloud'].endpoints.active_directory = 'https://login.microsoftonline.com'  # pylint: disable=line-too-long # change once active_directory is fixed in ARM for the public cloud
+        KNOWN_CLOUDS = list(cli_cloud_dict.values())
+except Exception as ex:  # pylint: disable=broad-except
+    print('The ARM_CLOUD_METADATA_URL environment variable points to an invalid cloud metadata URL')
```

Comment:
please use `logger.warn('Failed to load cloud metadata from the url specified by ARM_CLOUD_METADATA_URL')`

Reply:
done